internal/provider/resource_tfe_opa_version.go

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTFEOPAVersion() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEOPAVersionCreate,
+		Read:   resourceTFEOPAVersionRead,
+		Update: resourceTFEOPAVersionUpdate,
+		Delete: resourceTFEOPAVersionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTFEOPAVersionImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"sha": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"official": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"beta": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"deprecated": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"deprecated_reason": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  nil,
+			},
+		},
+	}
+}
+
+func resourceTFEOPAVersionCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+
+	opts := tfe.AdminOPAVersionCreateOptions{
+		Version:          *tfe.String(d.Get("version").(string)),
+		URL:              *tfe.String(d.Get("url").(string)),
+		SHA:              *tfe.String(d.Get("sha").(string)),
```

Comment:
Do you need to use pointers here? [See other example](https://github.com/hashicorp/terraform-provider-tfe/blob/main/internal/provider/resource_tfe_terraform_version.go#L71-L80).

Reply:
Yes, because the structs are different in go-tfe for [terraform-version](https://github.com/hashicorp/go-tfe/blob/main/admin_terraform_version.go#L74) and [opa/sentinel-version](https://github.com/hashicorp/go-tfe/blob/main/admin_terraform_version.go#L74). the tfe.String method only returns a pointer and need to be dereferenced